cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_data_resources.go

```diff
@@ -444,12 +443,38 @@ func (s *Server) inspectKappK8sResources(ctx context.Context, cluster, namespace
 		// TODO(minelson): We can't currently use `errors.IsNotFound(err)` here.
 		// See https://github.com/vmware-tanzu/carvel-kapp/issues/498
 		// Instead we need to match on the error string :/
-		if strings.Contains(err.Error(), fmt.Sprintf("configmaps %q not found", appName)) {
+		cmErrPattern := "configmaps %q not found"
+		appErrPattern := "App '%s' (namespace: %s) does not exist"
+		if strings.Contains(err.Error(), fmt.Sprintf(cmErrPattern, appName)) || strings.Contains(err.Error(), fmt.Sprintf(appErrPattern, appName, namespace)) || strings.Contains(err.Error(), fmt.Sprintf(appErrPattern, packageId, namespace)) {
```

Comment:
The messages differ from pre and post kapp 0.47:

- pre 0.47: just `configmaps...`
- post 0.47: `"App ...` using `packageId` for detecting the fallback mode and  `"App ...` using `appName` when the CM is certainly not found